laws/src/main/scala/cats/laws/discipline/MonadTests.scala

```diff
@@ -25,11 +26,18 @@ trait MonadTests[F[_]] extends ApplicativeTests[F] with FlatMapTests[F] {
     EqFABC: Eq[F[(A, B, C)]],
     EqFInt: Eq[F[Int]],
     iso: Isomorphisms[F]
-  ): RuleSet =
+  ): RuleSet = {
+    implicit def ArbFAA: Arbitrary[F[Either[A, A]]] =
+      Arbitrary(
+        Gen.oneOf(ArbFA.arbitrary.map(fa => laws.F.map(fa)(_.asLeft[A])),
+                  ArbFA.arbitrary.map(fa => laws.F.map(fa)(_.asRight[A]))
+        )
+      )
```

Comment:
We need `F[Either[A, A]]`, `F[Either[A, B]]`, `F[Boolean]`, `F[Unit]`, and maybe more to test all the selective laws, but for bincompat, we have what we have here.

Since we're a functor, we can make all those from what we have.  Is that the right approach?  And should we make them here and parameterize them on SelectiveTests, or should SelectiveLaws make their own from this set?

Reply:
I say we derive these from the ones passed in and not ask for more. I think we should always use a minimal set.

Reply:
I think that's a good idea.  It does make it harder to test against pure values.  For instance, if we derive `F[Boolean]` from `F[A]`, the `F[Boolean]` will be the result of a `map`.  If we request an `F[Boolean]`, we could get pure `F[Boolean]`s.  Per the functor laws, it shouldn't matter.  In more complex functors, like `IO` or `Parser`, a pure value and a mapped value may have different internal representations with different optimizations.  I think we'll leave it to the functor laws to detect violations there.